packages/cli-e2e/utils/processManager.ts

```diff
@@ -30,43 +29,13 @@ export class ProcessManager {
   };
 
   public async killAllProcesses() {
-    const promises: Promise<void>[] = [];
-    const processIterator = this.processes.values();
-    for (
-      let current = processIterator.next();
-      !current.done;
-      current = processIterator.next()
-    ) {
-      const currentProcess = current.value;
-      await new Promise<void>((resolve) => {
-        promises.push(
-          new Promise<void>((exit) => {
-            currentProcess.removeAllListeners('exit').on('exit', () => {
-              this.onExit(currentProcess)();
-              exit();
-            });
-            if (!Number.isInteger(currentProcess.pid)) {
-              console.error(
-                `Process pid is not a number. Received pid: ${currentProcess.pid}`
-              );
-              return resolve();
-            }
-            if (process.platform === 'win32') {
-              try {
-                recurseProcessKillWindows(currentProcess);
-              } catch (error) {
-                console.error(JSON.stringify({error}));
-              }
-            } else {
-              process.kill(-currentProcess.pid);
-            }
-            resolve();
-          })
-        );
-      });
-      current = processIterator.next();
-    }
-
-    await Promise.all(promises);
+    const pids: Array<number> = Array.from(this.processes.values())
+      .map(
+        process.platform === 'win32'
+          ? (process) => process.pid || 0
+          : (process) => -(process.pid || 0)
+      )
+      .filter((pid) => pid !== 0);
+    await fkill(pids, {tree: true, force: true, silent: true});
```

Comment:
> Aren't we already manually killing processes along with their parent process?
Yes, we were, this replaces that: "Replace the process killing custom logic with a reputable package."
>  will this throw an error?
No, `silent:true` make so that the operation do not fail/throw (especially if the process has closed itself gracefully before)